Sofa/framework/Core/src/sofa/core/objectmodel/BaseData.h

```diff
@@ -287,7 +280,7 @@ class SOFA_CORE_API BaseData : public DDGNode
     ///
     /// Note that this is a one-time copy and not a permanent link (otherwise see setParent())
     /// @return true if the copy was successful.
-    bool copyValueFrom(const BaseData* data);
+    virtual bool copyValueFrom(const BaseData* data);
```

Comment:
Adding the "virtual" keywords  breaks the delegation mechanism allowing the child class to not fullfill the parent API's. 

Given that  Data::copyValueFrom(const BaseData* data) is mimicking (with bug) the BaseData::copyValueFrom(const BaseData* data) behavior I would say that the proper way to clean that part is to totally remove Data::copyValueFrom(const BaseData* data).